torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1106,11 +1181,14 @@ def test_3_level_hierarchical_model_averager(self):
             averager = hierarchicalSGD.HierarchicalModelAverager(
                 period_group_size_dict=period_group_size_dict, warmup_steps=warmup_steps
             )
+            subgroup1 = averager.period_process_group_dict[subgroup_avg_period1]
+            subgroup2 = averager.period_process_group_dict[subgroup_avg_period2]
+
```

Comment:
Thanks for the refactoring! I guess the purpose of these changes is testing `period_process_group_dict`.

Can you directly add an assertion that `list(_pg_group_ranks[subgroup1].keys())` is equal to `range(subgroup_size1)` here? (Same for subgroup2).

Then the resulting list of either above expression can be reused in L1188 and L1191.

Reply:
fix. please cr again.

Reply:
Looks good. Thanks!